### PR TITLE
add test-utils and maven plugin helpers

### PR DESCRIPTION
**Pull request by MauricioUyaguari**

### CODE REVIEW

File: legend-sdlc-test-utils/pom.xml [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>com.gs.ti.ci.authn</groupId>
+                    <artifactId>gsauthnj-core</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>log4j</groupId>
+                    <artifactId>log4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-test-utils/pom.xml:94]: Are these exclusions necessary?

File: legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/project/maven/LegendFileGenerationPluginMavenHelper.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.finos.legend.sdlc.domain.model.version.VersionId;
+import org.finos.legend.sdlc.server.project.ProjectFileAccessProvider;
+
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+
+public class LegendFileGenerationPluginMavenHelper extends AbstractLegendMavenPluginHelper
+{
+
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/project/maven/LegendFileGenerationPluginMavenHelper.java:29]: Remove excess whitespace

File: legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/EntityValidator.java [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+
+public class EntityValidator
+{
+    private static final Pattern VALID_NAME_PATTERN = Pattern.compile("^\\w+$");
+    private static final Pattern VALID_PACKAGE_PATTERN = Pattern.compile("^\\w+(::\\w+)*$");
+    private static final Pattern VALID_CLASSIFIER_PATH_PATTERN = Pattern.compile("^meta(::\\w+)+$");
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/EntityValidator.java:41]: Use possessive quantifiers, if possible.

File: legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/junit/LegendSDLCTestSuiteBuilder.java [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    }
+
+    public TestSuite buildSuite(String name, EntityLoader entityLoader, ClassLoader classLoader)
+    {
+        return buildSuite(name, entityLoader.getAllEntities().collect(Collectors.toList()), classLoader);
+    }
+
+    public TestSuite buildSuite(String name, Collection<? extends Entity> entitiesForTesting, ClassLoader classLoader)
+    {
+        Map<String, TestSuiteBuilder> testSuiteBuilders = getTestSuiteBuilderByTypeMap(MappingTestCase.class, ServiceTestCase.class, DeprecatedServiceTestCase.class);
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/junit/LegendSDLCTestSuiteBuilder.java:105]: Let's replace this with a ServiceLoader approach. Then we can eliminate the need for the deprecated service test case here.

File: legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/junit/pure/v1/DeprecatedServiceTestCase.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.service.Service;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.service.ServiceTest;
+import org.finos.legend.pure.generated.Root_meta_pure_router_extension_RouterExtension;
+import org.finos.legend.sdlc.domain.model.entity.Entity;
+import org.finos.legend.sdlc.test.junit.LegendSDLCTestCase;
+import org.finos.legend.sdlc.test.junit.LegendSDLCTestCaseCollector;
+
+import java.util.function.Consumer;
+
+// Deprecated to account for previous classifier path of service
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/junit/pure/v1/DeprecatedServiceTestCase.java:31]: See my comment in LegendSDLCTestSuiteBuilder.

File: legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/junit/pure/v1/ServiceTestCase.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    {
+        super(pureModel, pureModelContextData, planTransformers, extensions, pureVersion, service);
+        this.serviceTest = serviceTest;
+        this.serviceTestRunner = new ServiceTestRunner(service, Tuples.pair(pureModelContextData, pureModel), this.planExecutor, extensions, planTransformers, pureVersion);
+    }
+
+    @Override
+    protected void doSetUp() throws Exception
+    {
+        super.doSetUp();
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/junit/pure/v1/ServiceTestCase.java:60]: Why is this method overridden?

**Review by kevin-m-knight-gs — CHANGES_REQUESTED** [review]

File: legend-sdlc-test-utils/src/test/java/org/finos/legend/sdlc/test/junit/TestLegendSDLCTestSuiteBuilder.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+
+public class TestLegendSDLCTestSuiteBuilder
+{
+    private static PureModel PURE_MODEL = new PureModel(PureModelContextData.newBuilder().build(), null, Thread.currentThread().getContextClassLoader(), DeploymentMode.PROD);
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-test-utils/src/test/java/org/finos/legend/sdlc/test/junit/TestLegendSDLCTestSuiteBuilder.java:36]: Is PURE_MODEL actually used?

File: legend-sdlc-test-utils/pom.xml [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-shared-core</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.zaxxer</groupId>
+                    <artifactId>HikariCP</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-test-utils/pom.xml:125]: Can we get rid of this dependency?

File: legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/junit/pure/v1/MappingTestCase.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        RichMappingTestResult richMappingTestResult = this.mappingTestRunner.doRunTest();
+        MappingTest mappingTest = this.mappingTestRunner.mappingTest;
+        if (richMappingTestResult.getResult() == TestResult.FAILURE)
+        {
+            fail("Test failure for mapping test '" + mappingTest.name + "' for Mapping '" + this.entity.getPath() + "'");
+        }
+        else if (richMappingTestResult.getResult() == TestResult.ERROR)
+        {
+
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/junit/pure/v1/MappingTestCase.java:67]: Eliminate excess blank line

**Review by kevin-m-knight-gs — CHANGES_REQUESTED** [review]

File: legend-sdlc-test-utils/pom.xml (hunk 8 above)
Comment (MauricioUyaguari) [reply legend-sdlc-test-utils/pom.xml]: `[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce) @ legend-sdlc-test-utils ---
[WARNING] 
Dependency convergence error for com.zaxxer:HikariCP:2.7.6 paths to dependency are:
+-org.finos.legend.sdlc:legend-sdlc-test-utils:0.33.1-SNAPSHOT
  +-org.finos.legend.sdlc:legend-sdlc-language-pure-compiler:0.33.1-SNAPSHOT
    +-org.finos.legend.engine:legend-engine-shared-core:2.29.1
      +-org.finos.legend.shared:legend-shared-pac4j-kerberos:0.11.1
        +-org.finos.legend.shared:legend-shared-pac4j:0.11.1
          +-org.pac4j:dropwizard-pac4j:3.0.0
            +-org.pac4j:pac4j-config:3.0.0
              +-com.zaxxer:HikariCP:2.7.6
and
+-org.finos.legend.sdlc:legend-sdlc-test-utils:0.33.1-SNAPSHOT
  +-org.finos.legend.engine:legend-engine-test-runner-service:2.29.1
    +-org.finos.legend.engine:legend-engine-executionPlan-execution-store-relational:2.29.1
      +-org.finos.legend.engine:legend-engine-executionPlan-execution-store-relational-connection:2.29.1
        +-com.zaxxer:HikariCP:3.4.5
[WARNING] Rule 1: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
Rule 1: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:


Failed while enforcing releasability. See above detailed error message.
`

**Review by kevin-m-knight-gs — APPROVED** [review]

**Pull request merged**